src/utils/NumberUtils.ts

```diff
@@ -0,0 +1,4 @@
+export const toValidPositiveInteger = (numericString: string) => {
+  const validInt = Number(numericString.replace(/[^0-9]+/g, ""));
+  return validInt < 1 ? 1 : validInt > 30 ? 30 : validInt;
```

Comment:
Shall we constantify `30`?

Reply:
Yes. It was a horrible idea to put it inside of this function because now the function needs to be called `toValidPositiveIntegerNotLargerThan30`  which is probably not what we want. Will create a validator function `toValidRepSetNumber` in `LiftUtils` and put 30 in there.